api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -593,6 +588,23 @@ public void mergePatients(Patient preferred, Patient notPreferred) throws APIExc
 		Context.getPersonService().savePersonMergeLog(personMergeLog);
 	}
 	
+	private void requireNoActiveOrderOfSameType(Patient patient1, Patient patient2) {
+		String messageKey = "Patient.merge.cannotHaveSameTypeActiveOrders";
+		List<Order> ordersByPatient1 = Context.getOrderService().getAllOrdersByPatient(patient1);
```

Comment:
Note that it's one thing to check if any of the patients has any active orders and it's another to copy over the orders, when checking if any of the patients has any active orders, getActiveOrders() is what you need to call, when moving over the orders to the preferred  patient, then you need to call getAllOrders, makes sense? In this method I believe you are checking if they have active orders and you should be calling getActiveOrder